lms/djangoapps/teams/tests/test_views.py

```diff
@@ -1769,6 +2050,59 @@ def test_expand_team(self):
         )
         self.verify_expanded_team(result['team'])
 
+    @ddt.data(
+        ('student_masters', 200),
+        ('student_masters_not_on_team', 200),
+        ('student_unenrolled', 404),
+        ('student_enrolled', 403),
+        ('student_enrolled_both_courses_other_team', 403),
+        ('staff', 200),
+    )
+    @ddt.unpack
+    def test_organization_protected(self, user, expected_status):
+        """ Users should not be able to see memberships for users in a different bubble than them """
+        self.get_membership_detail(
```

Comment:
student_enrolled and student_enrolled_both_courses_other_team should 404 rather than 403ing

- [ ] maybe add some more tests for combinations of nonexistant teams/users